sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -94,7 +103,11 @@ public void AddLink(string traceparent, string tracestate, IDictionary<string, s
 
         public void Start()
         {
-            _activityAdapter?.Start();
+            bool? started = _activityAdapter?.Start();
+            if (started.GetValueOrDefault() && _suppressNestedClientScopes)
```

Comment:
Are there any concurrency concerns here if two client methods are called from different threads?

Reply:
If two client methods are called in two different threads, they will end up having two different executions contexts (and two different `AsyncLocal` instances), so we should be fine here.

I've also had to change the underlying implementation to rely on [`Activity.SetCustomProperty`](https://github.com/dotnet/runtime/blob/ed50b63b404cec5ab03cff0b0fde1d2f7327751e/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L980) for other reasons, which is thread-safe